lib/galaxy/util/inflection.py

```diff
@@ -4,347 +4,116 @@
-        return self.Inflector.titleize(word, uppercase)
-
-    def camelize(self, word):
-        ''' Returns given word as CamelCased
-        Converts a word like "send_email" to "SendEmail". It
-        will remove non alphanumeric character from the word, so
-        "who's online" will be converted to "WhoSOnline"'''
-        return self.Inflector.camelize(word)
-
-    def underscore(self, word):
-        ''' Converts a word "into_it_s_underscored_version"
-        Convert any "CamelCased" or "ordinary Word" into an
-        "underscored_word".
-        This can be really useful for creating friendly URLs.'''
-        return self.Inflector.underscore(word)
-
-    def humanize(self, word, uppercase=''):
-        '''Returns a human-readable string from word
-        Returns a human-readable string from word, by replacing
-        underscores with a space, and by upper-casing the initial
-        character by default.
-        If you need to uppercase all the words you just have to
-        pass 'all' as a second parameter.'''
-        return self.Inflector.humanize(word, uppercase)
-
-    def variablize(self, word):
-        '''Same as camelize but first char is lowercased
-        Converts a word like "send_email" to "sendEmail". It
-        will remove non alphanumeric character from the word, so
-        "who's online" will be converted to "whoSOnline"'''
-        return self.Inflector.variablize(word)
-
-    def tableize(self, class_name):
-        ''' Converts a class name to its table name according to rails
-        naming conventions. Example. Converts "Person" to "people" '''
-        return self.Inflector.tableize(class_name)
-
-    def classify(self, table_name):
-        '''Converts a table name to its class name according to rails
-        naming conventions. Example: Converts "people" to "Person" '''
-        return self.Inflector.classify(table_name)
-
-    def ordinalize(self, number):
-        '''Converts number to its ordinal form.
-        This method converts 13 to 13th, 2 to 2nd ...'''
-        return self.Inflector.ordinalize(number)
-
-    def unaccent(self, text):
-        '''Transforms a string to its unaccented version.
-        This might be useful for generating "friendly" URLs'''
-        return self.Inflector.unaccent(text)
+        if number_of_records != 1:
+            return self.pluralize(word)
+        return word
 
-    def urlize(self, text):
-        '''Transform a string to its unaccented and underscored
-        version ready to be inserted in friendly URLs'''
-        return self.Inflector.urlize(text)
+    def _transform(self, rules, word, pluralize=True):
+        return self._handle_nonchanging(word) or self._handle_irregular(word, pluralize=pluralize) or self._apply_rule(rules, word) or word
 
-    def demodulize(self, module_name):
-        return self.Inflector.demodulize(module_name)
+    def _handle_nonchanging(self, word):
+        if word.lower() in self.NONCHANGING_WORDS:
```

Comment:
I think your modified version will pluralize `misinformation`, `subspecies` and so on, right?

Reply:
Oh yes, it will.. I can change that part back. The point it, computational linguistics is hairy and complex, and rules usually don't work very well. "heep" (not heap) was just one example I came up with that broke the suffix replacement approach. Your examples are definitely more relevant to our domain than "heep", so I'm, of course, fine with using the suffix approach. But, again, I think the overall algorithm is good enough in our case.

Reply:
I don't think the previous code was broken actually:
```python
        for uncountable_word in self.uncountable_words:
            if lower_cased_word[-1 * len(uncountable_word):] == uncountable_word:
                return word
```
When `uncountable_word` is `sheep` and `lower_cased_word` is `heep`, `lower_cased_word[-1 * len(uncountable_word):]` is `heep`, so the test would be correctly `False`.

A better version of the above test would be `lower_cased_word.endswith(uncountable_word)`

Reply:
Yes, you're right. I got it backwards. And yes, of course `endswith` is better.